File: app/src/main/java/com/numero/colorpalette/view/adapter/PagerAdapter.java
package com.numero.colorpalette.view.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.numero.colorpalette.fragment.ColorListFragment;
import com.numero.colorpalette.model.ColorStyle;

public class PagerAdapter extends FragmentPagerAdapter {

    private final Context context;

    public PagerAdapter(@NonNull Context context, @NonNull FragmentManager fragmentManager) {
        super(fragmentManager);
        this.context = context;
    }

    @Override
    public Fragment getItem(int position) {
        ColorStyle style = ColorStyle.values()[position];
        return ColorListFragment.newInstance(style.getColor());
    }

    @Override
    public int getCount() {
        return ColorStyle.values().length;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        ColorStyle style = ColorStyle.values()[position];
        return context.getString(style.getNameStringRes());
    }
}

File: app/src/main/java/com/numero/colorpalette/view/adapter/PagerAdapter.kt
package com.numero.colorpalette.view.adapter

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

import com.numero.colorpalette.fragment.ColorListFragment
import com.numero.colorpalette.model.ColorStyle

class PagerAdapter(private val context: Context, fragmentManager: FragmentManager) : FragmentPagerAdapter(fragmentManager) {

    private val colors: Array<ColorStyle> = ColorStyle.values()

    override fun getItem(position: Int): Fragment {
        val style = colors[position]
        return ColorListFragment.newInstance(style.color)
    }

    override fun getCount(): Int {
        return colors.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        val style = colors[position]
        return context.getString(style.nameStringRes)
    }
}
